### ISSUE TITLE
DLL problem while running catboost in VS  Code.

### ISSUE THREAD

**Issue by MartaLenson** [catboost/catboost#1473]
Problem:
catboost version: 0.24
Operating System: Windows 10

Hi,

I wrote a simple code with catboost in VS Code and run it once successfully.
Here is the code:
```
X_train, X_valid, y_train, y_valid = train_test_split(X, y, test_size=0.25, random_state=SEED)

params = {'loss_function':'MultiClass',
          'eval_metric':'AUC',
          'cat_features': cat_features,
          'verbose': 200,
          'random_seed': SEED
         }
model = CatBoostClassifier(**params)
model.fit(X_train, y_train,
          eval_set=(X_valid, y_valid),
          use_best_model=True,
          plot=True
         )

model.save_model('saved_model')
```
The next day I wanted to run it again and I got a DLL error:
```
UserWarning: To draw plots in fit() method you should install ipywidgets and ipython
  warnings.warn("To draw plots in fit() method you should install ipywidgets and ipython")
Traceback (most recent call last):
  File "\lib\site-packages\catboost\core.py", line 1004, in _get_catboost_widget
    from .widget import MetricVisualizer
  File "\lib\site-packages\catboost\widget\__init__.py", line 1, in <module>
    from .ipythonwidget import MetricVisualizer  # noqa
  File "\lib\site-packages\catboost\widget\ipythonwidget.py", line 6, in <module>
    from ipywidgets import DOMWidget, Layout, widget_serialization
  File "\lib\site-packages\ipywidgets\__init__.py", line 25, in <module>
    from .widgets import *
  File "\lib\site-packages\ipywidgets\widgets\__init__.py", line 4, in <module>
    from .widget import Widget, CallbackDispatcher, register, widget_serialization
  File "\lib\site-packages\ipywidgets\widgets\widget.py", line 18, in <module>
    from ipykernel.comm import Comm
  File "\lib\site-packages\ipykernel\__init__.py", line 2, in <module>
    from .connect import *
  File "\lib\site-packages\ipykernel\connect.py", line 18, in <module>
    import jupyter_client
  File "\lib\site-packages\jupyter_client\__init__.py", line 4, in <module>
    from .connect import *
  File "\lib\site-packages\jupyter_client\connect.py", line 21, in <module>
    import zmq
  File "\lib\site-packages\zmq\__init__.py", line 47, in <module>
    from zmq import backend
  File "\lib\site-packages\zmq\backend\__init__.py", line 40, in <module>
    reraise(*exc_info)
  File "\lib\site-packages\zmq\utils\sixcerpt.py", line 34, in reraise
    raise value
  File "\lib\site-packages\zmq\backend\__init__.py", line 27, in <module>
    _ns = select_backend(first)
  File "\lib\site-packages\zmq\backend\select.py", line 28, in select_backend
    mod = __import__(name, fromlist=public_api)
  File "\lib\site-packages\zmq\backend\cython\__init__.py", line 6, in <module>
    from . import (constants, error, message, context,
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "\execute.py", line 36, in <module>
    plot=True
  File "\lib\site-packages\catboost\core.py", line 4196, in fit
    silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model)
  File "\lib\site-packages\catboost\core.py", line 1788, in _fit
    with log_fixup(), plot_wrapper(plot, [_get_train_dir(self.get_params())]):
  File "\lib\contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "\lib\site-packages\catboost\core.py", line 1014, in plot_wrapper
    widget = _get_catboost_widget(train_dirs)
  File "\lib\site-packages\catboost\core.py", line 1008, in _get_catboost_widget
    raise ImportError(str(e))
**ImportError: DLL load failed: The specified module could not be found.**
```
The error message starts actuallly with a warning: To draw plots in fit() method you should install ipywidgets and ipython
However, I have both installed.

Is it ok to run the code in VS Code? Or it's only possible to run the code in jupyter notebooks?
As I mentioned before, yesterday the code run without any problems!

Thank you!

/Marta

**Comment by kizill** [comment]
Hello! It's not a bug of a CatBoost and it seems like a well known bug of some `pyzmq` versions:
https://stackoverflow.com/questions/54224969/import-error-while-trying-to-run-jupyter-notebook
https://github.com/ContinuumIO/anaconda-issues/issues/10422#issuecomment-448185070
https://github.com/nteract/hydrogen/issues/1517
https://github.com/nteract/hydrogen/issues/1405

TL;DR - uninstall and then reinstall `pyzmq` library.
